cola/cola.c

```diff
@@ -0,0 +1,102 @@
+#include <stdlib.h>
+#include "cola.h"
+
+#define COLA_VACIA 0
+
+struct nodo;
+typedef struct nodo nodo_t;
+
+struct nodo {
+    void *dato;
+    nodo_t *proximo;
+};
+
+nodo_t *nodo_crear(void *valor) {
+    nodo_t *nodo = malloc(sizeof(nodo_t));
+
+    if (!nodo) {
+        return false;
+    }
+
+    nodo->dato = valor;
+    nodo->proximo = NULL;
+    return nodo;
+}
+
+void nodo_destruir(nodo_t *nodo) {
+    free(nodo);
+}
+
+struct cola {
+    nodo_t *primero;
+    nodo_t *ultimo;
+};
+
+cola_t *cola_crear() {
+    cola_t *cola = malloc(sizeof(cola_t));
+    if (!cola) {
+        return NULL;
+    }
+
+    cola->primero = NULL;
+    cola->ultimo = NULL;
+    return cola;
+}
+
+bool cola_esta_vacia(const cola_t *cola) {
+    return !cola->primero || !cola->ultimo;
```

Comment:
Más que un or debería ser un and (porque si pasa una, tiene que pasar la otra). De todas formas, esto va a funcionar siempre.